rosbag2/test/rosbag2/rosbag2_write_integration_fixture.hpp

```diff
@@ -24,6 +24,8 @@
 #include "rclcpp/rclcpp.hpp"
 #include "rosbag2/rosbag2.hpp"
 #include "rosbag2_storage/serialized_bag_message.hpp"
+#include "rosbag2_storage_default_plugins/sqlite/sqlite_exception.hpp"
```

Comment:
I think this implies a strong dependency from `rosbag2` to `rosbag2_storage_default_plugins`. Ideally all these tests should perform equivally with all plugins.

We can introduce a new package for integration or system tests like these.

Reply:
In order to write integration tests for `rosbag2` we need a storage plugin. Which storage plugin is used for the tests is irrelevant and hidden inside the test fixtures.
Having `rosbag2_storage_default_plugins` as test dependency for `rosbag2` or having a separate test package depending on `rosbag2_storage_default_plugins` and `rosbag2` is for me practically equivalent. From a developers point of view I would prefer the test dependency as then the tests reside in the same package as the code they test.

When the `rosbag2_transport` package is introduced the tests will move there. If we choose to move the tests into a package of their own I suggest to do so after/ with the introduction of `rosbag2_transport`.